src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Extensions/Symbols/ITypeSymbolExtensions.cs

```diff
@@ -679,6 +679,14 @@ public static bool IsEnumType([NotNullWhen(true)] this ITypeSymbol? type, [NotNu
                 return false;
 
             default:
+                if (type is INamedTypeSymbol namedType && 
+                    namedType.SpecialType == SpecialType.None && // Not a special type
+                    namedType.TypeKind == TypeKind.Struct &&
+                    namedType.ContainingNamespace?.Name == "System" &&
+                    namedType.Name == "Guid")
```

Comment:
1. needs docs.  
2. can be more cleanly written as:


```suggestion
                    namedType is INamedType { Name: nameof(Guid), ContainingNamespace { Name: nameof(System), ContainingNamespace.IsGlobal: true } }
```